src/main/java/com/example/bookstoreapp/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.example.bookstoreapp.repository.impl;
+
+import com.example.bookstoreapp.model.Book;
+import com.example.bookstoreapp.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import java.util.List;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
+    @PersistenceContext
+    private EntityManager entityManager;
```

Comment:
let's stick to the `EntityManagerFactory` or `SessionFactory` as a field for now and create a new entity manager/session in each method, along with transaction, try-catch blocks and resources closure, just like we did before. We will modify our repositories in other tasks later on 😉